### DISCUSSION TITLE
Contextual modal routing

### DISCUSSION THREAD

**Discussion by theguriev** [nuxt/nuxt#22170] [Questions] [answered]
Can you guys please tell me how to implement contextual modal routing in Nuxt?
I need the background to remain the same and the modals to change depending on the router. Like in instagram
https://www.youtube.com/watch?v=4NR2uOBsIu4&t=11s

**Comment by Jannchie** [comment]
https://nuxt.com/docs/guide/directory-structure/layouts
You need to put the background in the `/layouts` and write the modal box part in the `/pages`.

**Reply by theguriev** [reply]
But the background can be any page. I need something like if the user go to /modal/foo the previous page is remaining as a background and the modal is opening on it.  It is possible if the modals will be handle by hash routing and the pages by the history but I'm looking more elegant solution.

**Comment by theguriev** [comment]
Here is a good example https://dev.to/devmdmamun/create-contextual-modal-navigation-with-react-router-v6-28k2 but on next :) 
https://dreamy-macaron-0c86da.netlify.app/

I'll try to do something similar but if you know how to do that properly on nuxt, just let me know plz

**Reply by Jannchie** [reply, marked as answer]
It may be that what you actually need is nesting routing.

try this: https://stackblitz.com/edit/github-tzeitj

**Reply by theguriev** [reply]
Thank you for the answer! Let me check :)

**Reply by Jannchie** [reply]
Here is some relevant documentation: https://nuxt.com/docs/guide/directory-structure/pages#nested-routes

**Comment by theguriev** [comment]
It is really close to what I need actually. But now I'm looking for the approach where I can open the modal everywhere. Because if I'll add foo/bar.vue in the current example it will not work because there is no child index/modal. Right?

**Comment by theguriev** [comment]
Looks like I'll have to use a `pages:extend` hook and add the children to every single rout what I have. 
<img width="513" alt="image" src="https://github.com/nuxt/nuxt/assets/8502021/2c453d8d-429b-41bc-8c16-1c113a8c9b57">

**Reply by theguriev** [reply]
I'll try this approach but I'm not sure that this is optimal.

**Reply by theguriev** [reply]
Yeeeeeey! Looks like this approach is working as expected. The code will be something like this 
<img width="632" alt="image" src="https://github.com/nuxt/nuxt/assets/8502021/abc3c13a-446a-4be0-976b-4a17b0724da9">
Thank you @Jannchie! You are the best! :) ❤️

If anyone find the more elegant solution, pleas tag me somewhere ;)

**Comment by theguriev** [comment]
Is this possible to do something similar in nuxt? https://nextgram.vercel.app/
I want to optimize the work with modals and it would be cool to be able to open a modal on the client with the backgroud of the previous page, but if you go directly to the link then do not render the backgroud ( prev page ). Is this possible? 
![NextGram](https://i.imgur.com/KF0KExk.gif)

**Reply by itsalimanuel** [reply]
Hello @theguriev ,I believe we can explore something like this! 😄


https://github.com/nuxt/nuxt/assets/44509661/c1891e55-0f98-4157-ae7c-5fcc49ca3adc



https://stackblitz.com/edit/nuxt-starter-5fbuih?file=app.vue

**Reply by mykyta-nikiforov** [reply]
@itsalimanuel thank you! One thing I see is that this solution breaks the browser "Back" button
